File: sol/solution/0001-0100/0009. Palindrome Number/Solution.kt
class Solution {
    fun isPalindrome(x: Int): Boolean {
        if (x < 0 || (x % 10 == 0 && x != 0)) return false
        var reversed = 0
        var num = x
        while (num > reversed) {
            reversed = reversed * 10 + num % 10
            num /= 10
        }
        return num == reversed || num == reversed / 10
    }
}

File: sol/solution/0001-0100/0009. Palindrome Number/Solution.rs
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        if x < 0 || (x % 10 == 0 && x != 0) {
            return false;
        }
        let mut reversed = 0;
        let mut num = x;
        while num > reversed {
            reversed = reversed * 10 + num % 10;
            num /= 10;
        }
        num == reversed || num == reversed / 10
    }
}